R/TunerIrace.R

```diff
@@ -0,0 +1,148 @@
+        ParamInt$new("nbConfigurations", default = 0, lower = 0),
+        ParamInt$new("mu", default = 5, lower = 1),
+        ParamInt$new("softRestart", default = 1, lower = 0, upper = 1),
+        ParamDbl$new("softRestartThreshold"),
+        ParamInt$new("digits", default = 4, lower = 1, upper = 15),
+        ParamFct$new("testType", default = "F-test",
+          levels = c("F-test", "t-test", "t-test-bonferroni", "t-test-holm")),
+        ParamInt$new("firstTest", default = 5, lower = 0),
+        ParamInt$new("eachTest", default = 1, lower = 1),
+        ParamDbl$new("confidence", default = 0.95, lower = 0, upper = 1),
+        ParamInt$new("capping", default = 0, lower = 0, upper = 1),
+        ParamFct$new("cappingType", default = "median", levels = c("median", "mean", "best", "worst")),
+        ParamFct$new("boundType", default = "candidate", levels = c("candidate", "instance")),
+        ParamDbl$new("boundMax", default = 0),
+        ParamInt$new("boundDigits", default = 0),
+        ParamDbl$new("boundPar", default = 1),
+        ParamDbl$new("boundAsTimeout", default = 1)
+      ))
+      ps$values = list(n_instances = 10)
+
+      super$initialize(
+        param_set = ps,
+        param_classes = c("ParamDbl", "ParamInt", "ParamFct", "ParamLgl"),
+        properties = c("dependencies", "single-crit"),
+        packages = "irace"
+      )
+    }
+  ),
+
+  private = list(
+    .optimize = function(inst) {
+      pv = self$param_set$values
+      terminator = inst$terminator
+      objective = inst$objective
+      
+      # Set resampling instances
+      ri = map(seq(pv$n_instances), function(n) {
+        r = objective$resampling$clone()
+        r$instantiate(objective$task)
+      })
+      pv$n_instances = NULL
+
+      # Make scenario
+      scenario = c(list(
+        targetRunner = targetRunner,
+        logFile = tempfile(),
+        instances = ri,
+        debugLevel = 0,
+        maxExperiments = if (class(inst$terminator)[1] == "TerminatorEvals") terminator$param_set$values$n_evals else 0,
+        maxTime = if (class(inst$terminator)[1] == "TerminatorRunTime") terminator$param_set$values$secs else 0,
+        targetRunnerData = list(inst = inst)
+      ), pv)
+
+      res = irace::irace(scenario = scenario, parameters = paradox_to_irace(inst$search_space))
+      
+      # Temporarily store result
+      private$.result_id = res$.ID.[1]
+    },
+
+    # The final configurations returned by irace are the elites of the final race.
+    # We store the best performing one.
+    # The reported performance value is the average of all resampling iterations.
+    .assign_result = function(inst) {
+      if(length(private$.result_id) == 0) {
+        stop("irace::irace did not return a result. The evaluated configurations are still accessible through the archive.")
+      }
+      id_configuration = NULL
+      res = inst$archive$data[id_configuration == private$.result_id, ]
```

Comment:
`get("id_configuration")` so you don't have to do this NULL thing

Reply:
I am always confused when `get()` works...